packages/text-field/icon/index.tsx

```diff
@@ -94,15 +108,27 @@ export default class Icon extends React.Component<
       removeAttr: (attr: keyof IconState) => (
         this.setState((prevState) => ({...prevState, [attr]: null}))
       ),
+      notifyIconAction: () => ( this.props.hasOwnProperty('onSelect')
+        ? this.props.onSelect!()
```

Comment:
First time I'm doing a review on my phone... Do you have a default prop set for onSelect? We can get rid of the hasOwnProperty if you set one to an empty function

Reply:
I think this might be an appropriate use for `hasOwnProperty`. Shortly I will be pushing a change where I eliminate it from the get tabidex(). 

I consciously chose against the approach of making it a defaultProp. Coincidentally, that is how I first went about it [0]. However, I ran into a problem where onSelect will only fire if tabindex != -1 due to pointer-events being set to none [1]. Since tabindex was not a required property for the passed icon I thought it more straightforward to add it based on if onSelect is present. If its has a default prop it will always be present and tabindex will always be initialized with 0 unless specified. This was the major factor in my decision to drop it as a default property.

I also seem to recall an issue in renderIcon from text-field where It was always passing an onSelect. Alas, I cannot quite recall the issue there and it was never committed. Maybe I was also setting a default prop for onLeadingIconSelect/onTrailingIconSelect?

Take your time on responding. A code review on a phone sounds painful.
 

[0] - https://github.com/material-components/material-components-web-react/pull/600/commits/df8627c239b047c58a3353743b2838369c689516#diff-78f850e429f25a98280e040459f8b54f

[1] https://github.com/material-components/material-components-web/blob/3cc4c6aa48ae8c1a307df542911e28a7808de41f/packages/mdc-textfield/icon/mdc-text-field-icon.scss#L33-L37

Reply:
That makes sense about the defaultProp...but we don't use hasOwnProperty anywhere else in the library, which makes me suspicious. Either we should be doing that everywhere, or we shouldn't do it here - it just breaks the pattern. If we decide to do it, we should instead open another bug to do it system wide.

Reply:
I hear you on  that mindset....I guess I could do something like the following. 

```typescript
{ notifyIconAction: (() =>  this.props.onSelect ? this.props.onSelect() : null) }
```

for some reason I seem to rely heavily on hasOwnProperty() in other projects. Old habits and all... Nevertheless, making this exact change doesn't seem to cause any breaking changes. Will this work for you?

Reply:
looks good to me!